streaming/src/main/scala/org/apache/spark/streaming/receiver/ReceiverSupervisorImpl.scala

```diff
@@ -136,14 +136,18 @@ private[streaming] class ReceiverSupervisorImpl(
       metadataOption: Option[Any],
       blockIdOption: Option[StreamBlockId]
     ) {
+    var rBlock = receivedBlock
     val blockId = blockIdOption.getOrElse(nextBlockId)
     val numRecords = receivedBlock match {
       case ArrayBufferBlock(arrayBuffer) => arrayBuffer.size
+      case IteratorBlock(iterator) => 
+        var arrayBuffer =  ArrayBuffer(iterator.toArray : _*)		
```

Comment:
I don't know that it's safe to copy the iterator contents into memory here. I don't think it's worth it for the UI.